### PR TITLE
PP-8947 refactor ChargeEntity to use FeeEntity list

### PR DESCRIPTION
**Pull request by SandorArpa**
## WHAT YOU DID
- refactor ChargeEntity to have a onetomany relationship with FeeEntitys
- refactor FeeEntity to have a manytoone relationship with ChargeEntity
- update test helpers and tests accordingly

with @nimalank7

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/fee/model/Fee.java [hunk 1]
```diff
@@ -17,12 +13,12 @@ private Fee(FeeType feeType, Long amount) {
         this.amount = amount;
     }
 
-    public static Fee of(FeeType feeType, Long amount) {
-        return new Fee(feeType, amount);
+    public static Fee of(Long amount) {
+        return new Fee(null, amount);
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/fee/model/Fee.java]: Looks like this constructor is only used in tests. What does having a null `FeeType` mean? Should these tests just set a `FeeType`?

File: src/test/java/uk/gov/pay/connector/charge/model/domain/ChargeEntityFixture.java [hunk 2]
```diff
@@ -52,7 +53,7 @@
     private SupportedLanguage language = SupportedLanguage.ENGLISH;
     private boolean delayedCapture = false;
     private Long corporateSurcharge = null;
-    private Long fee = null;
+    private List<Fee> fee;
```
Comment (stephencdaly) [inline src/test/java/uk/gov/pay/connector/charge/model/domain/ChargeEntityFixture.java]: Maybe we should make the variable name `fees` just to make it easier to visually parse the code?

File: src/main/java/uk/gov/pay/connector/charge/model/domain/ChargeEntity.java [hunk 3]
```diff
@@ -117,8 +117,8 @@
     @JoinColumn(name = "gateway_account_credential_id", updatable = false)
     private GatewayAccountCredentialsEntity gatewayAccountCredentialsEntity;
 
-    @OneToOne(mappedBy = "chargeEntity", fetch = FetchType.EAGER)
-    private FeeEntity fee;
+    @OneToMany(mappedBy = "chargeEntity")
+    private List<FeeEntity> fee = new ArrayList<>();
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/charge/model/domain/ChargeEntity.java]: Should this variable name be `fees` to make the code easier to visually parse?

**Review by stephencdaly — COMMENTED** [review]
Looks good, just a few style comments

File: src/test/java/uk/gov/pay/connector/charge/model/domain/ChargeEntityFixture.java (hunk 2 above)
Comment (SandorArpa) [reply src/test/java/uk/gov/pay/connector/charge/model/domain/ChargeEntityFixture.java]: 👍

File: src/main/java/uk/gov/pay/connector/charge/model/domain/ChargeEntity.java (hunk 3 above)
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/connector/charge/model/domain/ChargeEntity.java]: 👍

File: src/main/java/uk/gov/pay/connector/fee/model/Fee.java (hunk 1 above)
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/connector/fee/model/Fee.java]: FeeType null is for the current pricing. FeeType is only applicable for the new pricing.

Comment (stephencdaly) [reply src/main/java/uk/gov/pay/connector/fee/model/Fee.java]: Ahh, I see, but for current pricing it's explicitly passing in `null` using the other constructor which is why I was a bit confused. This is probably fine, as long as it's removed in the cleanup.

File: src/main/java/uk/gov/pay/connector/charge/model/domain/ChargeEntity.java [hunk 4]
```diff
@@ -455,12 +453,15 @@ public void setCorporateSurcharge(Long corporateSurcharge) {
         this.corporateSurcharge = corporateSurcharge;
     }
 
-    public void setFee(FeeEntity fee) {
-        this.fee = fee;
+    public void setFees(FeeEntity fees) {
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/connector/charge/model/domain/ChargeEntity.java]: The parameter should be called `fee` as it's a singular FeeEntity.

File: src/main/java/uk/gov/pay/connector/fee/model/Fee.java (hunk 1 above)
Comment (SandorArpa) [reply src/main/java/uk/gov/pay/connector/fee/model/Fee.java]: Already removed it :)

**Review by stephencdaly — APPROVED** [review]
LGTM!

**Pull request merged**